src/codeu/chat/common/Message.java

```diff
@@ -79,19 +81,17 @@ public int compareTo(Message msg) {
     return this.creation.compareTo(msg.creation);
   }
 
+
   @Override
   public int hashCode() {
-    final int prime = 31;
-    int result = 1;
-    result = prime * result + Integer.getInteger(id.toString());
-
-    return result;
+    System.out.println("Entered hashCode");
+    return hash(id);
   }
 
   @Override
   public boolean equals(Object o) {
```

Comment:
Need a better parameter name, because "o" is a little vague. I'm not sure what you are trying to compare.